src/goto-cc/armcc_cmdline.cpp

```diff
@@ -265,6 +265,21 @@ static const char *options_with_arg[]=
   nullptr
 };
 
+static bool
+prefix_in_list(const char *option, const char **list, std::string &prefix)
```

Comment:
I think we could simplify a lot of this if we stored the args in the mode as `std::vector<std::string>` instead of `char **`. Then we could also just use regular `std::find` or something like that instead of this.

Reply:
That sounds like a good follow up refactoring.

Reply:
Done.